Classes/OldModel/Event.php

```diff
@@ -1434,17 +1475,29 @@ public function getPaymentMethodsPlain(): string
             return '';
         }
 
-        $rows = \Tx_Oelib_Db::selectMultiple(
-            'title, description',
-            'tx_seminars_payment_methods, tx_seminars_seminars_payment_methods_mm',
-            'tx_seminars_payment_methods.uid = ' .
-            'tx_seminars_seminars_payment_methods_mm.uid_foreign ' .
-            'AND tx_seminars_seminars_payment_methods_mm.uid_local = ' .
-            $this->getTopicOrSelfUid() .
-            \Tx_Oelib_Db::enableFields('tx_seminars_payment_methods'),
-            '',
-            'tx_seminars_seminars_payment_methods_mm.sorting'
-        );
+        $queryBuilder = self::getQueryBuilderForTable('tx_seminars_payment_methods');
+        $rows = $queryBuilder
+            ->select('title', 'description')
+            ->from('tx_seminars_payment_methods')
+            ->from('tx_seminars_seminars_payment_methods_mm')
+            ->join(
+                'tx_seminars_payment_methods',
+                'tx_seminars_seminars_payment_methods_mm',
+                'mm',
+                $queryBuilder->expr()->eq(
+                    'tx_seminars_payment_methods.uid',
+                    $queryBuilder->quoteIdentifier('mm.uid_foreign')
+                )
+            )
+            ->where(
+                $queryBuilder->expr()->eq(
+                    'tx_seminars_seminars_payment_methods_mm.uid_local',
+                    $queryBuilder->createNamedParameter($this->getTopicOrSelfUid(), \PDO::PARAM_INT)
+                )
+            )
+            ->orderBy('mm.sorting')
+            ->execute()
+            ->fetchAll();
```

Comment:
Please let's have one private method `getPaymentMethodsAsArray`for getting the payment method data, and then `getPaymentMethodsPlain` and `getPaymentMethods` can use it.